packages/commerce-sdk-react/src/hooks/useMutation.ts

```diff
@@ -1,17 +1,24 @@
 /*
- * Copyright (c) 2022, Salesforce, Inc.
+ * Copyright (c) 2023, Salesforce, Inc.
  * All rights reserved.
  * SPDX-License-Identifier: BSD-3-Clause
  * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
  */
+import {MutationFunction, useMutation as useReactQueryMutation, UseMutationOptions} from '@tanstack/react-query'
 import useAuthenticatedClient from './useAuthenticatedClient'
-import {useMutation as useReactQueryMutation, UseMutationOptions} from '@tanstack/react-query'
-import {IMutationFunction} from './types'
+import {AddParameters, ApiClients} from './types'
 
-export const useMutation = <TData = unknown, TError = unknown, TVariables = unknown>(
-    fn: IMutationFunction<TData, TVariables>,
-    options?: Omit<UseMutationOptions<TData, TError, TVariables>, 'mutationFn'>
+type MutationFunctionWithApiClients<Data, Vars> = AddParameters<
+    MutationFunction<Data, Vars>,
+    // TODO: Remove this after merging in prettier v2 changes
+    // eslint-disable-next-line prettier/prettier
+    [apiClients: ApiClients]
+>
+
+export const useMutation = <Data, Err, Vars>(
```

Comment:
This was my first pass at cleaning up the existing types. I'll likely end up changing this a bit, as I did with `useQuery`.